packages/lesswrong/server/zohoUtils.ts

```diff
@@ -22,7 +22,14 @@ export const getEAGApplicationData = async (email: string) => {
     throw new Error('Failed to retrieve data from Zoho')
   }
   
-  // if no application data found in 2022, check 2021
+  // if no application data found in EAG 2022, check 2022 EAGx events
+  ['EAGxRotterdam', 'EAGxBerlin', 'EAGxSingapore'].forEach(async event => {
```

Comment:
forEach is forbidden for exactly this reason. This is async in forEach and will fail.

Reply:
Ah, sorry about that. In what sense is it "forbidden"?

Reply:
I think we should not use it. I've argued for it and maybe convinced people. I was maybe being overgenerous to my convincing abilities when I said "forbidden" as opposed to "JP thinks it's awful".